vault/resource_kv_secret_v2.go

```diff
@@ -134,8 +136,20 @@ func kvSecretV2Write(ctx context.Context, d *schema.ResourceData, meta interface
 		data[k] = d.Get(k)
 	}
 
-	if _, err := client.Logical().Write(path, data); err != nil {
-		return diag.Errorf("error writing secret data to %s, err=%s", path, err)
+	writeData := func() error {
```

Comment:
I don't believe we need this for the dedicated kvv2 resource. We shouldn't  encounter the migration issue going from kv1 to kv2.

Reply:
Thanks for taking a look, appreciate the feedback! I confirmed that the condition does indeed occur for both generic_secret and kv_secret_v2. In vault core, it looks like a kv_v2 mount is created as v1, then tuned to v2 and reloaded via reloadBackendCommon(), which returns without waiting for the upgrade to complete. I assume this is by design, so better to just retry on the terraform provider side...

Reply:
But in the case of new kv_v2 mount there would no data to migrate, no?